src/Adapter/Amp/EventLoop.php

```diff
@@ -55,26 +55,15 @@ public function async(\Generator $generator): Promise
                     }
                 }
             } catch (\Throwable $throwable) {
-                $fnFailure($throwable);
+                $deferred->getPromiseWrapper()->throwIfNotYielded($throwable);
```

Comment:
Is it still necessary since this test is done in destructor?
If not, I think we can remove `hasBeenYielded` and `throwIfNotYielded` functions.
 (same remark in other adapters)